src/actix/auth.rs

```diff
@@ -126,33 +126,46 @@ where
             return Box::pin(self.service.call(req));
         }
 
-        // Grab API key from request
-        let key =
-            // Request header
-            req.headers().get("api-key").and_then(|key| key.to_str().ok()).map(|key| key.to_string())
-                // Fall back to authentication header with bearer token
-                .or_else(|| {
-                    Authorization::<Bearer>::parse(&req).ok().map(|auth| auth.as_ref().token().into())
-                });
-
-        if let Some(key) = key {
-            let is_allowed = if let Some(ref auth_keys) = self.auth_keys {
-                auth_keys.can_write(&key) || (is_read_only(&req) && auth_keys.can_read(&key))
-            } else {
-                // This code path should not be reached
-                log::warn!("Auth for REST API is set up incorrectly. Denying access by default.");
-                false
-            };
-            if is_allowed {
+        let invalid_api_key = |req: ServiceRequest| -> Self::Future {
+            Box::pin(async {
```

Comment:
Can this be its own function?

Reply:
It would require too much type boilerplate...
```rust
fn invalid_api_key<S, B>(
    req: ServiceRequest,
) -> LocalBoxFuture<'static, Result<ServiceResponse<EitherBody<S::Response, BoxBody>>, Error>>
where
    S: Service<ServiceRequest, Response = ServiceResponse<EitherBody<B, BoxBody>>, Error = Error>,
    S::Future: 'static,
    B: 'static,
{
    Box::pin(async {
        Ok(req
            .into_response(HttpResponse::Forbidden().body("Invalid api-key"))
            .map_into_right_body())
    })
}
```

Reply:
🤯 Ok